crates/db_views/src/registration_applications/registration_application_view.rs

```diff
@@ -13,98 +14,90 @@ use lemmy_db_schema::{
   aliases,
   newtypes::{PersonId, RegistrationApplicationId},
   schema::{local_user, person, registration_application},
-  utils::{get_conn, limit_and_offset, DbConn, DbPool, ListFn, Queries, ReadFn},
+  utils::{get_conn, limit_and_offset, DbPool},
 };
 
-enum ReadBy {
-  Id(RegistrationApplicationId),
-  Person(PersonId),
+#[diesel::dsl::auto_type]
+fn joins() -> _ {
+  registration_application::table
+    .inner_join(local_user::table.on(registration_application::local_user_id.eq(local_user::id)))
+    .inner_join(person::table.on(local_user::person_id.eq(person::id)))
+    .left_join(
+      aliases::person1
+        .on(registration_application::admin_id.eq(aliases::person1.field(person::id).nullable())),
+    )
 }
 
-fn queries<'a>() -> Queries<
-  impl ReadFn<'a, RegistrationApplicationView, ReadBy>,
-  impl ListFn<'a, RegistrationApplicationView, RegistrationApplicationQuery>,
-> {
-  let all_joins = |query: registration_application::BoxedQuery<'a, Pg>| {
-    query
-      .inner_join(local_user::table.on(registration_application::local_user_id.eq(local_user::id)))
-      .inner_join(person::table.on(local_user::person_id.eq(person::id)))
-      .left_join(
-        aliases::person1
-          .on(registration_application::admin_id.eq(aliases::person1.field(person::id).nullable())),
-      )
-      .order_by(registration_application::published.desc())
-      .select((
-        registration_application::all_columns,
-        local_user::all_columns,
-        person::all_columns,
-        aliases::person1.fields(person::all_columns).nullable(),
-      ))
-  };
-
-  let read = move |mut conn: DbConn<'a>, search: ReadBy| async move {
-    let mut query = all_joins(registration_application::table.into_boxed());
-
-    query = match search {
-      ReadBy::Id(id) => query.filter(registration_application::id.eq(id)),
-      ReadBy::Person(person_id) => query.filter(person::id.eq(person_id)),
-    };
-
-    query.first(&mut conn).await
-  };
-
-  let list = move |mut conn: DbConn<'a>, o: RegistrationApplicationQuery| async move {
-    let mut query = all_joins(registration_application::table.into_boxed());
-
-    // If viewing all applications, order by newest, but if viewing unresolved only, show the oldest
-    // first (FIFO)
-    if o.unread_only {
-      query = query
-        .filter(registration_application::admin_id.is_null())
-        .order_by(registration_application::published.asc());
-    } else {
-      query = query.order_by(registration_application::published.desc());
-    }
-
-    if o.verified_email_only {
-      query = query.filter(local_user::email_verified.eq(true))
-    }
-
-    let (limit, offset) = limit_and_offset(o.page, o.limit)?;
-
-    query = query.limit(limit).offset(offset);
-
-    query.load::<RegistrationApplicationView>(&mut conn).await
-  };
+#[diesel::dsl::auto_type]
+/// A missing admin id, means the application is unread
+fn is_unread() -> _ {
+  registration_application::admin_id.is_null()
+}
```

Comment:
You can even extract simple filters, which I've done here since its used in several places.